Compilation error C2220 with warning C4267 on msvc with _UNICODE UNICODE

I use Visual Studio 17 2022 generator with this msvc:
```
Microsoft (R) C/C++ Optimizing Compiler Version 19.44.35211 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```

When I set cmake compile definitions `add_compile_definitions(UNICODE _UNICODE)`, then I get compilation error:
```
1>...\libs\sources\benchmark\src\sysinfo.cc(464,31): error C2220: the following warning is treated as an error
1>...\libs\sources\benchmark\src\sysinfo.cc(464,31): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
```

Minimal cmake project:

[google-bench-issue.zip](https://github.com/user-attachments/files/21100143/google-bench-issue.zip)

--- a/src/sysinfo.cc
+++ b/src/sysinfo.cc
@@ -461,7 +461,7 @@ std::string GetSystemName() {
                                 DWCOUNT, NULL, 0, NULL, NULL);
   str.resize(len);
   WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, hostname, DWCOUNT, &str[0],
-                      str.size(), NULL, NULL);
+                      static_cast<int>(str.size()), NULL, NULL);
 #endif
   return str;
 #elif defined(BENCHMARK_OS_QURT)

